rainforest/rfml.go

```diff
@@ -263,3 +269,120 @@ func (r *RFMLWriter) WriteRFMLTest(test *RFTest) error {
 
 	return nil
 }
+
+// ParseEmbeddedFiles replaces file step variable paths with values expected
+// by Rainforest. eg: {{ file.screenshot(my_screenshot.gif) }} would be translated
+// to the format {{ file.screenshot(FILE_ID, FILE_SIGNATURE) }}.
+func (c *Client) ParseEmbeddedFiles(test *RFTest) error {
+	if test.TestID == 0 {
+		return fmt.Errorf("Cannot parse embedded files without a test ID.")
+	}
+
+	uploadedFiles, err := c.getUploadedFiles(test.TestID)
+	if err != nil {
+		return err
+	}
+
+	digestToFileMap := map[string]uploadedFile{}
+	for _, f := range uploadedFiles {
+		digestToFileMap[f.Digest] = f
+	}
+
+	replaceEmbeddedFilePaths := func(text string, embeddedFiles []embeddedFile) (string, error) {
+		out := text
+		for _, embed := range embeddedFiles {
+			filePath := embed.path
+			if strings.HasPrefix(filePath, "~/") {
+				var usr *user.User
+				usr, err = user.Current()
+				if err != nil {
+					return "", err
+				}
+				filePath = filepath.Join(usr.HomeDir, filePath[2:])
+			} else if test.RFMLPath == "" {
+				return "", fmt.Errorf("Cannot parse relative file path %v for RFML test %v. RFMLPath field cannot be blank.", filePath, test.RFMLID)
+			} else {
+				rfmlDirectory := filepath.Dir(test.RFMLPath)
+				filePath = filepath.Join(rfmlDirectory, filePath)
+			}
+
```

Comment:
if you are using `filepath.Abs` you don't really need to do `~/` and `rfmlDirectory` joins above. Just set `rfmlDirectory` as a working directory and Abs should do all of the magicks for you.